plugins/permCheck.ts

```diff
@@ -19,21 +26,296 @@
+	return CommandControlPlugin<CommandType.Both>(async (ctx) => {
 		if (ctx.guild === null) {
-			await ctx.reply("This command cannot be used here");
-			console.warn(
-				"PermCheck > A command stopped because we couldn't check a users permissions (was used in dms)",
-			); //delete this line if you dont want to be notified when a command is used outside of a guild/server
+			await ctx.reply("This command cannot be used in DM's!");
+			return controller.stop();
+		}
+		if (
+			!(ctx.member! as GuildMember)
+				.permissionsIn(ctx.channel as TextChannel)
+				.has(perm)
+		) {
+			await ctx.reply({
+				embeds: [
+					sendEmbed(
+						response ??
+							`You are missing required permissions to run this command:\n${permsToString(
+								perm
+							)}`
+					),
+				],
+				ephemeral: !ctx.isMessage() ? true : false,
+			});
 			return controller.stop();
 		}
-		if (!(ctx.member! as GuildMember).permissions.has(perm)) {
-			await ctx.reply(response);
+		if (
+			!(ctx.member! as GuildMember)
+				.permissionsIn(ctx.channel as TextChannel)
+				.any(perm)
+		) {
+			await ctx.reply({
+				embeds: [
+					sendEmbed(
+						response ??
+							`You need at least one of the following permissions to run this command:\n${permsToString(
+								perm
+							)}`
+					),
+				],
+				ephemeral: ctx.isMessage() ? false : true,
+			});
 			return controller.stop();
 		}
 		return controller.next();
 	});
 }
+function subGroups(opts: BaseOptions[]) {
+	return CommandControlPlugin<CommandType.Slash>(async ({ interaction }) => {
+		await no_guild(interaction);
+		const member = interaction.member as GuildMember;
+		const group = interaction.options.getSubcommandGroup();
+		for (const opt of opts) {
+			if (group !== opt.name) {
+				await interaction.reply({
+					embeds: [
+						sendEmbed(
+							`[PLUGIN_permCheck.subGroups]: Failed to find specified subcommandGroup \`${opt.name}\`!`
+						),
+					],
+					ephemeral: true,
+				});
+				return controller.stop();
+			}
+			const _perms = member.permissionsIn(
+				interaction.channel as TextChannel
+			);
+			if (opt.needAllPerms && !_perms.has(opt.perms)) {
+				await interaction.reply({
+					embeds: [
+						sendEmbed(
+							opt.response ??
+								`You cannot use this group due to missing permissions: ${permsToString(
+									opt.perms
+								)}`
+						),
+					],
+					ephemeral: true,
+				});
+				return controller.stop();
+			}
+			if (!opt.needAllPerms && !_perms.any(opt.perms)) {
+				await interaction.reply({
+					embeds: [
+						sendEmbed(
+							opt.response ??
+								`You cannot use this group because you need at least one of the following permissions: ${permsToString(
+									opt.perms
+								)}`
+						),
+					],
+					ephemeral: true,
+				});
+				return controller.stop();
+			}
+		}
+		return controller.next();
+	});
+}
+
+function subcommands(opts: BaseOptions[]) {
+	return CommandControlPlugin<CommandType.Slash>(async ({ interaction }) => {
```

Comment:
same issue, destructuring the interaction here will make this plugin probably fail for Both Commands that run with a text command.